Array sniffs documentation (Related to #1722)

Mike-Hermans opened the pull request:
Adds documentation for the following sniffs:
- Arrays.MultipleStatementAlignment
- Arrays.ArrayKeySpacingRestrictions

Related to #1722

GaryJones commented on WordPress/Docs/Arrays/ArrayKeySpacingRestrictionsStandard.xml:
```diff
@@ -0,0 +1,23 @@
+    ]]>
+    </standard>
+    <code_comparison>
+        <code title="Correct whitespace for index keys">
```
Start with `Valid: `.

GaryJones commented on WordPress/Docs/Arrays/ArrayKeySpacingRestrictionsStandard.xml:
```diff
@@ -0,0 +1,23 @@
+$post_title = $post['post_title'];
+        ]]>
+        </code>
+        <code title="Incorrect spacing">
```
Start with `Invalid: `.

GaryJones commented on WordPress/Docs/Arrays/MultipleStatementAlignmentStandard.xml:
```diff
@@ -0,0 +1,29 @@
+    ]]>
+    </standard>
+    <code_comparison>
+        <code title="Double arrow operators aligned">
```
Start with `Valid: `.

GaryJones commented on WordPress/Docs/Arrays/MultipleStatementAlignmentStandard.xml:
```diff
@@ -0,0 +1,29 @@
+);
+        ]]>
+        </code>
+        <code title="Not aligned; harder to read">
```
Start with `Invalid: `.

jrfnl commented:
@Mike-Hermans Just wondering if you'll have a chance to finish this off in the near future.
I'd like to release WPCS 2.2.0 soon and it would be great if this PR could be included.

If you haven't got time or lost interest, please let us know and we'll see if we can find someone to take over.

Mike-Hermans commented:
I've updated the examples in ArrayKeySpacingRestrictions to include both a concatenated string and a constant.

For MultipleStatementAlignment I've added another block with code comparision to show the correct spacing in single line arrays, and the multiple spaces for alignment in multi-line arrays.

jrfnl commented on WordPress/Docs/Arrays/MultipleStatementAlignmentStandard.xml:
```diff
@@ -0,0 +1,44 @@
+    <code_comparison>
+        <code title="Valid: correct spacing between the key and value.">
+        <![CDATA[
+$args( 'cat'<em> => </em>22, 'year'<em> => </em>$current_year );
```
Multi-item associative arrays are not allowed to be single-line according to the WP Coding Standards, so while this sniff doesn't address that, the example does not comply with the other coding standards.

It also isn't an array declaration the way it is now (it's actually a parse error).

May I suggest splitting this and having two code samples for both `valid` as well as `invalid` for this block ?

```xml
        <code title="Valid: correct spacing between the key and value.">
        <![CDATA[
$foo = array( 'cat'<em> => </em>22 );
$bar = array( 'year'<em> => </em>$current_year );
        ]]>
        </code>
        <code title="Invalid: No or incorrect spacing between the key and value.">
        <![CDATA[
$foo = array( 'cat'<em>=></em>22 );
$bar = array( 'year'<em>=>   </em>$current_year );
        ]]>
        </code>
```

jrfnl commented on WordPress/Docs/Arrays/MultipleStatementAlignmentStandard.xml:
```diff
@@ -0,0 +1,44 @@
+    </code_comparison>
+    <standard>
+    <![CDATA[
+      In the case of a block of related assignments, more space may be inserted to promote readability.
```
If you're updating the file anyway, I think making this one a bit less like a _mild suggestion_ would be good idea.

```suggestion
      In the case of a block of related assignments, it is recommended to align the arrows to promote readability.
```
